doc/source/glossary.rst

```diff
@@ -0,0 +1,188 @@
+    possible to use them as a way to map a particular address within a (potentially larger, but sparsely populated)
+    virtual space to a concrete (and usually contiguous) physical space, through the process of :ref:`mapping<map>`.
+
+.. _Pointer:
+
+Pointer
+    A value within memory that points to a different area of memory.  This allows objects to contain references to
+    other objects without containing all the data of the other object.  Following a pointer is known as :ref:`dereferencing<dereference>`
+    a pointer.  Pointers are usually the same length as the maximum address of the address space, since they
+    should be able to point to any address within the space.
+
+R
+-
+.. _Range:
+
+Range
+    This is the set of the possible output values for a mapping or mathematical function.
+
+S
+-
+.. _Struct:
+
+Struct, Structure
+    A means of containing multiple different :ref:`type<types>` associated together.  A struct typically contains
+    other :ref:`type<types>`, usually :ref:`aligned<alignment>` (unless :ref:`packing<packed>` is involved).  In this way
+    the :ref:`members<member>` of a type can be accessed by finding the data at the relative :ref:`offset<offset>` to
+    the start of the structure.
+
+.. _Symbol:
+
+Symbol
+    This is used in many different contexts, as a short term for many things.  Within Volatility, a symbol is a
+    construct that usually encompasses a specific type :ref:`type<Type>` at a specfific :ref:`offset<Offset>`,
+    representing a particular instance of that type within the memory of a compiled and running program.  An example
+    would be the location in memory of a list of active tcp endpoints maintained by the networking stack
+    within an operating system.
+
+T
+-
+.. _Template:
+
+Template
+    Within volatility 3, the term template applies to a :ref:`type<Type>` that has not yet been instantiated or linked
+    to any data or a specific location within memory.  Once a type has been tied to a particular chunk of data, it is
+    called an :ref:`object<Object>`.
+
+.. _Translation Layer:
+
+Translation Layer
+    This is a specific type of :ref:`data layer<Data Layer>`, a non-contiguous group of bytes that can be references by
+    a unique :ref:`offset<Offset>` within the layer.  In particular, translation layers translates (or :ref:`maps<Map>`)
+    requests made of it to a location within a lower layer.  This can be either linear (a one-to-one mapping between bytes)
+    or non-linear (a group of bytes :ref:`maps<Map>` to a larger or smaller group of bytes.
+
+.. _Type:
+
+Type
+    This is a structure definition of multiple elements that expresses how data is laid out.  Basic types define how
+    the data should be interpretted in terms of a run of bits (or more commonly a collection of 8 bits at a time,
+    called bytes).  More complex types can be made up of other types combined together at specific locations known
```

Comment:
Consider: `specific offsets` instead of `specifc locations`. 
Sentence structure made me trip, suggesting:

New types can be made up by combining other types at specific locations, known as structs, or by repeating the same type, known as arrays. 

Next sentence is missing some words:
They can even be defined as multiple types at the same location, depending on .....